tests/test_cli.py

```
@@ -1291,5 +1326,1849 @@ def test_handle_auth_logout_prints_header(self, mock_print):
+        # Import and call function
+        from mmrelay.cli import _validate_credentials_json
+        result = _validate_credentials_json(config_path)
+
+        # Verify results
+        self.assertFalse(result)
+        # Should report all missing fields
+        mock_print.assert_any_call("❌ Error: credentials.json missing required fields: access_token, user_id, device_id")
+        mock_print.assert_any_call("   Please run 'mmrelay auth login' again to generate new credentials that include a device_id.")
+
+
+class TestAuthLogout(unittest.TestCase):
```

Ruff (0.12.2)

Redefinition of unused `TestAuthLogout` from line 1120

(F811)